modules/orchestrator/initialize.go

```diff
@@ -65,15 +86,16 @@ func Initialize() error {
 	server.WithLocaleLoader(bdl.GetLocaleLoader())
 	// server.Router().Path("/metrics").Methods(http.MethodGet).Handler(promxp.Handler("orchestrator"))
 	server.RegisterEndpoint(ep.Routes())
+	p.server = server
 
-	if err := initCron(ep); err != nil {
-		return err
-	}
-
-	logrus.Infof("start the service and listen on address: \"%s\"", conf.ListenAddr())
-	logrus.Errorf("[alert] starting orchestrator instance")
+	// Limit only one instance of scheduler to do the cron jobs
+	p.Election.OnLeader(func(ctx context.Context) {
+		logrus.Infof("i'm the leader now")
+		_ = initCron(ep, ctx)
+		logrus.Infof("i resign the leader now")
```

Comment:
Do you need invoke Election.ResignLeader here?

Reply:
Nope. The resign login has been done by the p.Election internally

Reply:
The ctx is closed after the resign.